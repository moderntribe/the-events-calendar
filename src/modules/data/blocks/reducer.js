/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { setInitialState as datetimeSetInitialState } from './datetime';
import organizers, { setInitialState as organizersSetInitialState } from './organizers';
import price, { setInitialState as priceSetInitialState } from './price';
import website, { setInitialState as websiteSetInitialState } from './website';
import venue, { setInitialState as venueSetInitialState } from './venue';

export const setInitialState = ( data ) => {
	datetimeSetInitialState( data );
	organizersSetInitialState( data );
	priceSetInitialState( data );
	venueSetInitialState( data );
	websiteSetInitialState( data );
};

export default combineReducers( {
	datetime,
	venue,
	organizers,
	price,
	website,
} );
