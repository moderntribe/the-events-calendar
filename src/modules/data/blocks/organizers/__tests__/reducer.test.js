/**
 * Internal dependencies
 */
import reducer, { setInitialState } from '@moderntribe/events/data/blocks/organizers/reducer';

const data = {
	meta: {
		_EventOrganizerID: [ 99, 100 ],
	},
};

describe( '[STORE] - Organizers reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toMatchSnapshot();
	} );

	it( 'Should set the initial state', () => {
		setInitialState( data );
		expect( reducer( undefined, {} ) ).toMatchSnapshot();
	} );
} );
