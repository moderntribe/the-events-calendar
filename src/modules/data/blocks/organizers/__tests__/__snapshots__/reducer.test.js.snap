// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Organizers reducer Should return the default state 1`] = `
Object {
  "blocks": Object {
    "allIds": Array [],
    "byId": Object {},
  },
  "classic": Array [],
}
`;

exports[`[STORE] - Organizers reducer Should set the initial state 1`] = `
Object {
  "blocks": Object {
    "allIds": Array [],
    "byId": Object {},
  },
  "classic": Array [
    99,
    100,
  ],
}
`;
